src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -98,8 +99,11 @@ export class PmdCatalogWrapper extends PmdSupport {
 		const pathSetMap = new Map<string, Set<string>>();
 
 		const customPathEntries: Map<string, Set<string>> = await this.getCustomRulePathEntries();
+		const fileHandler = new FileHandler();
+
+
 		// Iterate through the custom paths.
-		customPathEntries.forEach(async (paths: Set<string>, langKey: string) => {
+		for (const [langKey, paths] of customPathEntries) {
```

Comment:
`forEach` doesn't work correctly with `async` code. This blog post has more information https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404 I'm not sure how the code on line 108 was succeeding before, there may have been a race condition that we never noticed.

Before changing this, I was seeing indeterminate behavior when testing. I looked through the code to see if we need to fix it anywhere else, I didn't find any.

Reply:
Great find! However, I always get a bit nervous about applying `for-of` to objects instead of iterators. Could we use `customPathEntries.entries()` instead? We could rename the variable to `customPathMap` so we're not calling `xEntries.entries()`.

Reply:
Good point, will change.